rest-service/manager_rest/rest/filters_utils.py

```diff
@@ -1,129 +1,138 @@
+                raise BadFilterRule(
+                    filter_rule,
+                    f"The operator for filtering by attributes must be one"
+                    f" of {', '.join(ATTRS_OPERATORS)}")
+            if filter_rule_key not in resource_model.allowed_filter_attrs:
+                raise BadFilterRule(filter_rule, err_attr_msg)
 
         else:
-            raise BadLabelsFilter(labels_filter)
-
-    return labels_filters_mapping
-
-
-def _parse_labels_filter(labels_filter, sign):
-    """Validate and parse a labels filter
-
-    :param labels_filter: One of <key>=<value>, <key>=[<value1>,<value2>,...],
-           <key>!=<value>, <key>!=[<value1>,<value2>,...]
-    :param sign: Either '=' or '!='
-    :return: The labels_filter, with its key and value(s) in lowercase and
-             stripped of whitespaces
-    """
-    try:
-        raw_label_key, raw_label_value = labels_filter.split(sign)
-    except ValueError:  # e.g. a=b=c
-        raise BadLabelsFilter(labels_filter)
-
-    label_key = raw_label_key.strip().lower()
-    label_values = _get_label_value(raw_label_value.strip().lower())
-    value_msg_prefix = (None if len(label_values) == 1 else
-                        'One of the filter values')
-
-    parsed_label_values_list = []
-    for value in label_values:
-        try:
-            parsed_value = value.strip()
-            validate_inputs({'filter key': label_key})
-            validate_inputs({'filter value': parsed_value},
-                            err_prefix=value_msg_prefix)
-        except manager_exceptions.BadParametersError as e:
-            err_msg = 'The filter rule {0} is invalid. '.format(labels_filter)
-            raise manager_exceptions.BadParametersError(err_msg + str(e))
-
-        parsed_label_values_list.append(parsed_value)
-
-    return label_key, parsed_label_values_list
-
-
-def _get_label_value(raw_label_value):
-    if raw_label_value.startswith('[') and raw_label_value.endswith(']'):
-        return raw_label_value.strip('[]').split(',')
-
-    return [raw_label_value]
+            raise BadFilterRule(filter_rule,
+                                f"Filter rule type must be one of "
+                                f"{', '.join(FILTER_RULE_TYPES)}")
+
+        value_msg_prefix = (None if len(filter_rule_values) == 1 else
+                            'One of the filter rule values')
+
+        parsed_values_list = []
+        for value in filter_rule_values:
+            try:
+                validate_inputs({'filter rule key': filter_rule_key})
+                validate_inputs({'filter rule value': value},
+                                err_prefix=value_msg_prefix)
+            except BadParametersError as e:
+                err_msg = f'The filter rule {filter_rule} is invalid. '
+                raise BadParametersError(err_msg + str(e))
+            parsed_values_list.append(value.lower())
+
+        parsed_filter_rules.append({'key': filter_rule_key.lower(),
```

Comment:
perhaps this should just be a class, rather than a dict with always the same 4 keys

I note that even in tests you have a `_get_filter_rule_from_params` that would otherwise be unnecessary

Reply:
Yep, you're absolutely right. I'll change it to be a class.